packages/adapters/subgraph/src/lib/operations/queries.ts

```diff
@@ -294,72 +306,98 @@ export const getOriginTransfersQuery = (agents: Map<string, SubgraphQueryMetaPar
   `;
 };
 
-const originByXCalledTimestampQueryString = (
+const originTransferByNonceQueryString = (
   prefix: string,
-  originDomain: string,
-  fromTimestamp: number,
+  fromNonce: number,
   maxBlockNumber?: number,
   orderDirection: "asc" | "desc" = "desc",
 ) => {
   return `${prefix}_originTransfers(
     where: {
-      originDomain: ${originDomain},
-      timestamp_gte: ${fromTimestamp},
+      nonce_gte: ${fromNonce},
       ${maxBlockNumber ? `, blockNumber_lte: ${maxBlockNumber}` : ""}
     },
-    orderBy: timestamp,
+    orderBy: nonce,
     orderDirection: ${orderDirection}
   ) {${ORIGIN_TRANSFER_ENTITY}}`;
 };
 
-export const getOriginTransfersByXCallTimestampQuery = (
-  params: Map<string, SubgraphQueryByTimestampMetaParams>,
-): string => {
+export const getOriginTransfersByNonceQuery = (agents: Map<string, SubgraphQueryMetaParams>): string => {
   const { config } = getContext();
 
   let combinedQuery = "";
   const domains = Object.keys(config.sources);
   for (const domain of domains) {
     const prefix = config.sources[domain].prefix;
-    if (params.has(domain)) {
-      combinedQuery += originByXCalledTimestampQueryString(
+    if (agents.has(domain)) {
+      combinedQuery += originTransferByNonceQueryString(
         prefix,
-        domain,
-        params.get(domain)!.fromTimestamp,
-        params.get(domain)!.maxBlockNumber,
-        params.get(domain)!.orderDirection,
+        agents.get(domain)!.latestNonce,
+        agents.get(domain)!.maxBlockNumber,
+        agents.get(domain)!.orderDirection,
       );
     } else {
       console.log(`No agents for domain: ${domain}`);
     }
   }
 
   return gql`
-    query GetOriginTransfersByXCallTimestamp {
+    query GetOriginTransfers {
         ${combinedQuery}
       }
   `;
 };
 
-const destinationTransfersByExecuteTimestampQueryString = (
+const destinationTransferByNonceQueryString = (
   prefix: string,
-  fromTimestamp: number,
-  maxBlockNumber?: number,
+  fromNonce: number,
   orderDirection: "asc" | "desc" = "desc",
 ) => {
-  return `
-  ${prefix}_destinationTransfers(
-    where: { 
-      executedTimestamp_gte: ${fromTimestamp}, 
-      ${maxBlockNumber ? `, executedBlockNumber_lte: ${maxBlockNumber}` : ""} 
-    }, 
-    orderBy: executedTimestamp, 
+  return `${prefix}_destinationTransfers(
+    where: {
+      nonce_gte: ${fromNonce},
```

Comment:
The main problem we run into here is that the reconciled transfers don't have a nonce. They only have a timestamp.

Reply:
True, we will not get reconciled but not executed transfers here.
They would be fetched by the reconciled timestamp queries, separately

We could use timestamps everywhere, but using nonce is more robust, as it handles the case where there are more transfers that got executed in 1 second, than can fit in response, and cause us repeatedly process then same transfers and get stuck.

So, thought use nonce when we can, if not use timestamp.

Reply:
This makes sense and is likely the best approach for now.